src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -6183,80 +6152,84 @@ static void Main()
         }
 
         [Fact]
-        public void UnderscoreLocalInDeconstructDeclaration()
+        public void EscapedUnderscoreInDeclarationCSharp9()
         {
             var source =
 @"
 class C
 {
     static void Main()
     {
-        int _;
-        (_, var x) = (1, 2);
-        System.Console.Write(_);
+        (@_, var x) = (1, 2);
     }
 }
 ";
 
-            var comp = CreateCompilation(source, options: TestOptions.DebugExe);
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe, parseOptions: TestOptions.Regular9);
             comp.VerifyDiagnostics(
-                // (7,9): error CS8184: A deconstruction cannot mix declarations and expressions on the left-hand-side.
-                //         (_, var x) = (1, 2);
-                Diagnostic(ErrorCode.ERR_MixedDeconstructionUnsupported, "(_, var x)").WithLocation(7, 9)
+                // (6,9): error CS8652: The feature 'Mixed declarations and expressions in deconstruction' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         (@_, var x) = (1, 2);
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "(@_, var x) = (1, 2)", isSuppressed: false).WithArguments("Mixed declarations and expressions in deconstruction").WithLocation(6, 9),
+                // (6,10): error CS0103: The name '_' does not exist in the current context
+                //         (@_, var x) = (1, 2);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "@_", isSuppressed: false).WithArguments("_").WithLocation(6, 10)
                 );
 
             var tree = comp.SyntaxTrees.First();
             var model = comp.GetSemanticModel(tree);
 
-            var discard = GetDiscardIdentifiers(tree).First();
-            Assert.Equal("(_, var x)", discard.Parent.Parent.ToString());
-            var symbol = (ILocalSymbol)model.GetSymbolInfo(discard).Symbol;
-            Assert.Equal("int _", symbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat));
-            Assert.Equal("System.Int32", model.GetTypeInfo(discard).Type.ToTestDisplayString());
+            Assert.Empty(GetDiscardIdentifiers(tree));
         }
 
         [Fact]
-        public void ShortDiscardInDeconstructAssignment()
+        public void UnderscoreLocalInDeconstructDeclaration()
         {
             var source =
 @"
 class C
 {
     static void Main()
     {
-        int x;
-        (_, _, x) = (1, 2, 3);
-        System.Console.Write(x);
+        int _;
+        (_, var x) = (1, 2);
+        System.Console.Write(_);
     }
 }
 ";
 
-            var comp = CreateCompilation(source, options: TestOptions.DebugExe);
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe, parseOptions: TestOptions.RegularPreview);
             comp.VerifyDiagnostics();
-            CompileAndVerify(comp, expectedOutput: "3");
+            CompileAndVerify(comp, expectedOutput: "1");
```

Comment:
Let's VerifyIL for the tests that actually are expected to run.

Reply:
Sorry, this was a more general comment for the tests below as well

---
In reply to: [490632506](https://github.com/dotnet/roslyn/pull/44476#discussion_r490632506) [](ancestors = 490632506)

Reply:
@YairHalberstadt I think this is my only comment left outstanding.

Reply:
I'm now verifying IL for all tests I added modified, except a couple @jcouv said not to because they're too long.

There is a test directly below which github thinks I modified, but that's just the git diff going wrong.